src/solvers/particle_swarm.jl

```diff
@@ -297,7 +297,15 @@ function particle_swarm!(
     social_weight::Real=1.4,
     cognitive_weight::Real=1.4,
     stopping_criterion::StoppingCriterion=StopAfterIteration(500) |
-                                          StopWhenChangeLess(1e-4),
+                                          StopWhenEntryChangeLess(
+        :swarm,
+        (p, st, old_swarm, swarm) -> distance(
+            PowerManifold(get_manifold(p), NestedPowerRepresentation(), length(swarm)),
+            old_swarm,
+            swarm,
+        ),
+        1e-4,
+    ),
```

Comment:
This looks like a very odd criterion. Why can't you just stop when the change in optimal point is less than 1e-4?

Reply:
That is actually the same as before, just that before we (sometimes) handled the swarm as the iterate and had an iterate change.

The reason to not check the (now) iterates change is, that even in the first iteration, this one might not change, while the swarm moves, maybe even all other points, depending on the social and velocity parameters.
Only if they also settle to move less it is very likely to indicate a reason to stop.

Reply:
Then it would be easier to calculate the norm of velocities? There would be no need for (sometimes expensive) distance calculation and remembering the old swarm.

Reply:
The reference we provide even just has a maxiter thing, so I would have to check how we came up with that one. But the iterate change one will indicate to stop too early, even an adapted one to check for the populations best ones.

Reply:
Velocities sounds reasonable as well; but if we change this, then I would like to check literature before again a bit.

Reply:
Sure, if this just replicates the old behavior in a nicer way, we can consider changing this stopping criterion in another PR.

Reply:
Yes, this is exactly the same behaviour as before. To provide some detail:

Previously the swarm was sometimes considered to the the iterate (changed that in #344 ), what I did not yet change there is the one place where we still to some extend considered the swarm the iterate, since the `StopWhenChangeLess` was still redefined (until today) to check the change of the swarm instead.

With the new criterion this redefinition was changed to really use the Entry Change stopping criterion.

So for that reason these lines reproduce the behaviour from before, just in a better fitting stopping criterion name and with a nicer/shorter form.